### PR TITLE
Feat: Add Individual Component Imports

### PR DESCRIPTION
**Pull request by tkottke90**
# Description

Related to #35.  I started importing Kobalte into my Solid project and noticed that there was a significant (8-10 second based on a Google Lighthouse score) delay before my page had loaded.  

image

Looking at the developer tools (specifically the network tab in Firefox) I could see that there was about 3.54 MB of files loaded into the browser.  Based on the file names I could see that _every_ Kobalte module was being loaded regardless of how many were being used:

image

Naturally I was curious about bundle sizes as some were in the related issue.  I was happy to see this package was tree-shakeable

image

This drew me to the conclusion that it was simply a developer experience issue.  The cost of having to re-load the entirety of Kobalte on any non-cached reload meant seeing a white screen while all of the resources downloaded to the browser.

# Solution

Seeing the open issue and the conservation that was going on it appeared to me that this could be resolved by adding sub-folder exports to the core module:

```tsx
import Dialog from "../../kobalte/dialog";

export function App() {

  return (
    <Dialog.Root>
      ...
    </Dialog.Root>
  );
}
```

To achieve this a sub-folder export was added to the `package.json` file which allowed each of the component directories to be targeted directly:

```json
// package.json
{
  ...,
  exports: {
    ...,
    "./*": {
      "types": "./dist/types/*/index.d.ts",
      "solid": "./dist/source/*/index.jsx"
    }
  }
}
```

One of the goals I had when working on this project was to ensure backwards compatibility.  To this effort a `default` export was added to each component module could be a "drop-in replacement" for the current version of Kobalte.  This way consumers of this library could transition when they were ready or the Kobalte team can decide on if/when deprecation of the previous pattern would occur.

### CODE REVIEW

**Comment by Murzbul** [comment]
What would be the problem with this feature? The issue of downloading and improving performance seems quite important to me.

**Comment by fabien-ml** [comment]
Hi, sorry for the late response. I've no problem with the PR, I just didn't had time to review it yet.

**Comment by tkottke90** [comment]
Resolved the merge conflicts that were being reported so those don't block the merge

File: packages/core/src/pagination/index.tsx [hunk 1]
```diff
@@ -38,4 +38,6 @@ export type {
   PaginationRootOptions,
   PaginationRootProps,
 };
-export { Ellipsis, Item, Items, Next, Previous, Root };
+
+export { Ellipsis, Item, List, Next, Previous, Root };
```
Comment (fabien-ml) [inline packages/core/src/pagination/index.tsx:42]: `PaginationList` aka `List` is now `PaginationItems` aka `Items`

**Comment by thomas-evans** [comment]
@tkottke90 @fabien-ml Bumping: chrome_MiXbenOyZk
Thank you all for the work on this, if I can help please message me.

**Comment by aminya** [comment]
Is this work still in progress? What is remaining to allow this to work? I am interested in helping this merge.

File: packages/core/src/accordion/index.tsx [hunk 2]
```diff
@@ -27,3 +27,4 @@ export type {
   AccordionTriggerProps,
 };
 export { Content, Header, Item, Root, Trigger };
+export default { Content, Header, Item, Root, Trigger };
```
Comment (aminya) [inline packages/core/src/accordion/index.tsx:30]: Exposing default exports is not necessary for this PR. The `package.json` change should be enough.

**Comment by jer3m01** [comment]
Integrated this in #391 and changed the API.
Thanks for your PR!

**Pull request closed without merging**